File: docs/data-tools/codesnippet/VisualBasic/save-data-in-a-transaction_3.vb
    Private Sub DeleteCustomers()

        Dim deletedCustomers As NorthwindDataSet.CustomersDataTable
        deletedCustomers = CType(NorthwindDataSet.Customers.GetChanges(Data.DataRowState.Deleted),
            NorthwindDataSet.CustomersDataTable)

        If Not IsNothing(deletedCustomers) Then
            Try
                CustomersTableAdapter.Update(deletedCustomers)

            Catch ex As Exception
                MessageBox.Show("DeleteCustomers Failed" & vbCrLf & ex.Message)
            End Try
        End If
    End Sub

File: docs/data-tools/codesnippet/CSharp/save-data-in-a-transaction_3.cs
        private void DeleteCustomers()
        {
            NorthwindDataSet.CustomersDataTable deletedCustomers;
            deletedCustomers = (NorthwindDataSet.CustomersDataTable)
                northwindDataSet.Customers.GetChanges(DataRowState.Deleted);

            if (deletedCustomers != null)
            {
                try
                {
                    customersTableAdapter.Update(deletedCustomers);
                }
                catch (System.Exception ex)
                {
                    MessageBox.Show("DeleteCustomers Failed");
                }
            }
        }